packages/manager/src/features/Domains/DomainDetail/DomainRecords/TruncatedWithTooltip.tsx

```diff
@@ -0,0 +1,27 @@
+import { Tooltip } from '@linode/ui';
+import React from 'react';
+
+interface TruncatedWithTooltipProps {
+  maxLength?: number;
+  text: string;
+}
+
+const truncateEnd = (maxLength: number, text: string): string =>
+  text.length <= maxLength ? text : `${text.slice(0, maxLength)}…`;
+
+export const TruncatedWithTooltip = ({
+  maxLength = 50,
+  text,
+}: TruncatedWithTooltipProps) => {
+  const truncated = truncateEnd(maxLength, text);
+
+  if (text.length > maxLength) {
+    return (
+      <Tooltip title={text}>
+        <span>{truncated}</span>
+      </Tooltip>
+    );
+  }
+
+  return <>{truncated}</>;
```

Comment:
I think we can put the `truncateEnd` inside the `if` so that it only runs if necessary

I also think it is a bit more intuitive to return `text` if no truncation is necessary rather than `truncated`

```suggestion
  if (text.length > maxLength) {
    const truncated = truncateEnd(maxLength, text);

    return (
      <Tooltip title={text}>
        <span>{truncated}</span>
      </Tooltip>
    );
  }

  return text;
```